src/services/User.service.ts

```diff
@@ -44,3 +44,15 @@ export const updateUsersStatusWhoNeedsPasswordReset = async (
   );
   return updateUsersRequest;
 };
+
+export const updateUserService = async (
+  userId: string,
+  updatedUserData: UserAttributes,
+) => {
+  const userToUpdate = await User.findByPk(userId);
+  if (!userToUpdate) {
+    throw new Error(`User with id ${userId} not found`);
+  }
+  await userToUpdate.update(updatedUserData);
```

Comment:
`  const result = await User.update(updatedUserData,
    { where: { id: userId } }
  )`

make one request to the database, also remember to verify the data you are storing to the database, look at joi

Reply:
thank you a lot